ibc-clients/ics07-tendermint/src/client_state/execution.rs

```diff
@@ -339,12 +340,9 @@ where
         let host_timestamp = ctx.host_timestamp()?.into_tm_time();
 
         let tm_consensus_state_timestamp = tm_consensus_state.timestamp();
-        let tm_consensus_state_expiry = (tm_consensus_state_timestamp
-            + client_state.trusting_period)
-            .map_err(|_| ClientError::Other {
-                description: String::from(
-                    "Timestamp overflow error occurred while attempting to parse TmConsensusState",
-                ),
+        let tm_consensus_state_expiry =
+            (tm_consensus_state_timestamp + client_state.trusting_period).map_err(|e| {
+                DecodingError::invalid_raw_data(format!("failed to parse TmConsensusState: {e}"))
```

Comment:
This isn't actually a `DecodingError`. it's caused by the timestamp overflowing.

Reply:
I just realized there's a `Timestamp` error type. Should we opt to remove this similarly to how we removed `HeightError`?

Reply:
Due to the fact that `TimestampError` type is currently living in the `ibc-primitives` crate, we cannot consolidate this type into the module-level error types until #1327 is completed.